### PR TITLE
refactoring Xyk to Market class (mintLiquidity)

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/parallel/proofOfStake.3rdPartyRewards.activation.test.ts [hunk 1]
```diff
@@ -136,6 +127,18 @@ describe("Proof of stake tests", () => {
...
+      Sudo.sudoAs(
+        testUser3,
+        Market.mintLiquidity(
+          liqId,
+          GASP_ASSET_ID,
+          Assets.DEFAULT_AMOUNT,
+          Assets.DEFAULT_AMOUNT.muln(2),
+        ),
+      ),
+    );
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.test.ts:141]: can you do this before, so the mintLiquidity happen in the same batch?  or perhaps the pool is created in the same batch?

File: test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /*
  *
  * @group xyk
+ * @group market
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts:4]: only this test with this group? or u are using it for debugging?

File: test/parallel/xyk-pallet.asset.buy.test.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /*
  *
  * @group xyk
+ * @group market
```
Comment (goncer) [inline test/parallel/xyk-pallet.asset.buy.test.ts:4]: only thi schange on this file?? :sweat_smile:

File: testPad/Bug_burnTokens.test.ts [hunk 4]
```diff
@@ -136,7 +136,8 @@ describe("staking - testpad", () => {
 
     await api!.tx.utility
       .batch([
-        api!.tx.xyk.createPool(
+        api!.tx.market.createPool(
+          "Xyk",
           GASP_ASSET_ID,
```
Comment (goncer) [inline testPad/Bug_burnTokens.test.ts:141]: remove evth from testpad directly,

File: testPad/CleanBootstrap.ts [hunk 5]
```diff
@@ -75,7 +75,7 @@ function difference(a1: any, a2: any) {
   const tokens = await api!.query.tokens.accounts(user.address, liqToken);
   await signTx(
     api!,
-    api!.tx.xyk.burnLiquidity(0, 7, tokens.free.toString()),
+    api!.tx.market.burnLiquidity(liqToken, tokens.free.toString(), 0, 0),
```
Comment (goncer) [inline testPad/CleanBootstrap.ts:78]: remove this file too, do not waste your time

File: utils/market.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import { BN } from "@polkadot/util";
+import { api, Extrinsic } from "./setup";
+
+export class Market {
+  static createPool(
+    firstAsset: BN,
+    firstAssetAmount: BN,
+    secondAsset: BN,
+    secondAssetAmount: BN,
+    kind: any = "Xyk",
```
Comment (goncer) [inline utils/market.ts:10]: this will be an enum, but its fine for now :)

File: utils/setupsOnTheGo.ts [hunk 7]
```diff
@@ -2105,23 +2146,23 @@ async function findMerkleRange(
...
+  testLog.getLog().info(`root: ${root}`);
+
+  const range = await publicClient.readContract({
+    address: getL1(network)!.contracts.rollDown.address,
+    abi: abi,
+    functionName: "merkleRootRange",
+    args: [root],
+    blockTag: "latest",
+  });
+  return range as [bigint, bigint];
```
Comment (goncer) [inline utils/setupsOnTheGo.ts:2165]: ty for linting!

**Review by goncer — APPROVED** [review]

File: test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.API.fees.tripleTokens.test.ts]: I change all xyk tags to market in files where I delete xyk link

File: test/parallel/xyk-pallet.asset.buy.test.ts (hunk 3 above)
Comment (AlexChetverov) [reply test/parallel/xyk-pallet.asset.buy.test.ts]: No-no, sure, this is the change from the last PR. Fixed

File: utils/market.ts (hunk 6 above)
Comment (AlexChetverov) [reply utils/market.ts]: I see, I'll try to remmeber it

File: testPad/Bug_burnTokens.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply testPad/Bug_burnTokens.test.ts]: Done

File: testPad/CleanBootstrap.ts (hunk 5 above)
Comment (AlexChetverov) [reply testPad/CleanBootstrap.ts]: Done

**Review by goncer — APPROVED** [review]

**Pull request merged**